diff -urN a2d46031d4-78929cf818.clean/sources/pyside6/cmake/Macros/PySideModules.cmake a2d46031d4-78929cf818/sources/pyside6/cmake/Macros/PySideModules.cmake
--- a2d46031d4-78929cf818.clean/sources/pyside6/cmake/Macros/PySideModules.cmake	2025-03-24 23:19:07.000000000 +0100
+++ a2d46031d4-78929cf818/sources/pyside6/cmake/Macros/PySideModules.cmake	2025-05-08 16:22:38.012124654 +0200
@@ -338,7 +338,7 @@
     if(NOT (PYSIDE_IS_CROSS_BUILD OR DISABLE_PYI))
         set(generate_pyi_options ${module_NAME} --sys-path
             "${pysidebindings_BINARY_DIR}"
-            "${SHIBOKEN_PYTHON_MODULE_DIR}/..")     # use the layer above shiboken6
+            "${SHIBOKEN_PYTHON_MODULE_DIR}/../..")     # use the layer above shiboken6
         if (QUIET_BUILD)
             list(APPEND generate_pyi_options "--quiet")
         endif()
